src/main/java/org/skriptlang/skript/bukkit/recipes/MutableRecipe.java

```diff
@@ -0,0 +1,411 @@
+package org.skriptlang.skript.bukkit.recipes;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.*;
+import org.bukkit.inventory.recipe.CookingBookCategory;
+import org.bukkit.inventory.recipe.CraftingBookCategory;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.recipes.RecipeUtils.RecipeType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Mutable Recipe class allowing data to be set before creation of a recipe.
+ */
+public abstract class MutableRecipe implements Recipe {
+
+	private final RecipeType recipeType;
+	private final NamespacedKey key;
+	private ItemStack result = null;
+	private final List<String> errors = new ArrayList<>();
+
+	public MutableRecipe(NamespacedKey key, RecipeType recipeType) {
+		this.key = key;
+		this.recipeType = recipeType;
+	}
+
+	public void setResult(ItemStack result) {
+		this.result = result;
+	}
+
+	@Override
+	public @Nullable ItemStack getResult() {
```

Comment:
the method you're overriding is marked as @NotNull, why can result be null?

Reply:
I only did that because all the checks within the create methods were underlined saying "result will never be null"

Reply:
it should be marked as undefined nullability